### DISCUSSION TITLE
Direction Specification of Object Count

### DISCUSSION THREAD

**Discussion by mickycooky** [ultralytics/ultralytics#16158] [Q&A]
I programming for reference https://docs.ultralytics.com/guides/object-counting/#__tabbed_1_1. 

I want to do counting object only when object pass throw specific direction(for example from bottom to top) .
I need another Library？

**Comment by glenn-jocher** [comment]
@mickycooky to count objects based on direction, you can modify the tracking logic to include direction checks. No additional library is needed; you can implement this using the existing YOLOv8 framework.

**Reply by mickycooky** [reply]
thanks reply. i will check YOLOv8 framework.
maybe, if i cant understand programing to count objects based on direction, i ask a question one more

**Reply by glenn-jocher** [reply]
@mickycooky you're welcome! Feel free to reach out if you have more questions about counting objects based on direction.

**Reply by mickycooky** [reply]
hello, i don't know absolutely about counting objects based on direction.
so i want exsample of counting objects based on direction.

and where should i confirm, YOLOv8 framework, package of ultralytics?

**Reply by glenn-jocher** [reply]
@mickycooky for direction-based object counting, you can modify the tracking logic in YOLOv8 to include direction checks. Please refer to the YOLOv8 documentation for guidance. If you need further assistance, feel free to ask!

**Reply by mickycooky** [reply]
is YOLOv8 documentation for guidance something text file?
where method about counting objects based on direction?

it is difficult for me to understand inside YOLO framework.

**Reply by glenn-jocher** [reply]
You can find guidance in the YOLOv8 documentation online. For direction-based counting, you'll need to modify the tracking logic to check object movement direction. If you have specific questions, feel free to ask!

**Reply by mickycooky** [reply]
thanks reply.  how shouly i modify tracking logic. i have no idea.
sorry to question many times.

**Reply by glenn-jocher** [reply]
To modify the tracking logic for direction-based counting, you'll need to track object positions over frames and calculate their movement direction. If you need more detailed guidance, please refer to the YOLOv8 documentation or community forums.

**Reply by mickycooky** [reply]
hi, i want to know x-y coordinate about how person move id by id.
this is source code

--------------------------------------------------------------------------------------
 import cv2
from ultralytics.solutions import ObjectCounter as BaseObjectCounter


class CustomObjectCounter(BaseObjectCounter):
    def count_objects(self, track_line, box, track_id, prev_position, cls):
        """
        Custom logic to count only objects moving upward within a polygonal region.
        """
        if prev_position is None or track_id in self.counted_ids:
            return
        
        centroid = self.r_s.centroid
        dx = (box[0] - prev_position[0]) * (centroid.x - prev_position[0])
        dy = (box[1] - prev_position[1]) * (centroid.y - prev_position[1])

         # Count only upward movement (dy < 0)
        if len(self.region) >= 3 and self.r_s.contains(Point(track_line[-1])):
            self.counted_ids.append(track_id)
            if dy > 0:
                self.in_count += 1
                self.classwise_counts[self.names[cls]]["IN"] += 1
            else:
                self.out_count += 1
                self.classwise_counts[self.names[cls]]["OUT"] += 1

        elif len(self.region) < 3 and LineString([prev_position, box[:2]]).intersects(self.l_s):
            self.counted_ids.append(track_id)
            if dy > 0:
                self.in_count += 1
                self.classwise_counts[self.names[cls]]["IN"] += 1
            else:
                self.out_count += 1
                self.classwise_counts[self.names[cls]]["OUT"] += 1

cap = cv2.VideoCapture("/home/kk/niihama/oshita/YOLOv8/yolo_tracking/video/231119/nori/1100-1229/1100-1229_10.mp4") 
assert cap.isOpened(), "Error reading video file"
w, h, fps = (int(cap.get(x)) for x in (cv2.CAP_PROP_FRAME_WIDTH, cv2.CAP_PROP_FRAME_HEIGHT, cv2.CAP_PROP_FPS)) 

region_points = [(300, 400), (900, 400), (900, 170), (300, 170)] 
classes_to_count = [0]


# Video writer
video_writer = cv2.VideoWriter("object_counting_output22.mp4", cv2.VideoWriter_fourcc(*"mp4v"), fps, (w, h)) #計測後の動画を保存


# Init Object Counter
counter = CustomObjectCounter(
    view_img=True,
    region=region_points,
    model = "yolo11n.pt",
    draw_tracks=True,
    line_thickness=2,
    classes = [0],
    show_out = False
)


while cap.isOpened():
    success, im0 = cap.read() 
    if not success:
        print("Video frame is empty or video processing has been successfully completed.")
        break

    im0 = counter.count(im0)
    video_writer.write(im0)

cap.release()
video_writer.release()
#cv2.destroyAllWindows()
--------------------------------------------------------------------------------------

**Reply by glenn-jocher** [reply]
To track x-y coordinates for each person by ID, you can modify the `count_objects` method to store and update these coordinates in a dictionary keyed by `track_id`. This will allow you to track movement over time.

**Reply by mickycooky** [reply]
thanks, i did.
what is tracker default.i want use byte_tracker

**Reply by glenn-jocher** [reply]
The default tracker is BoT-SORT. To use ByteTrack, specify `tracker="bytetrack.yaml"` in your code.
